packages/shared/src/hooks/useTagAndSource.ts

```diff
@@ -0,0 +1,119 @@
+import { useContext } from 'react';
+import AuthContext from '../contexts/AuthContext';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+import useMutateFilters from './useMutateFilters';
+import { Source } from '../graphql/sources';
+import AlertContext from '../contexts/AlertContext';
+
+interface TagActionArguments {
+  tags: Array<string>;
+  category?: string;
+}
+
+interface SourceActionArguments {
+  source: Source;
+}
+
+export default function useTagAndSource({
```

Comment:
All the subfunctions here are missing showing login in case the user is not signed in except the follow tags.
We can use the origin prop as the first argument for the `showLogin` func.

Let's also provide the `postId` on every tracked event here, just in case. Btw, our convention in the analytics system is an underscore case. So please change to `post_id: postId`

Reply:
Ah didn't add the call as it's virtually impossible to trigger these without being logged in.
But a good fallback! will do 🤘